aspnetcore/razor-components/index.md

```diff
@@ -80,44 +52,85 @@ When this component is used elsewhere in the app, IntelliSense speeds developmen
+
+Because Razor Components decouple a component's rendering logic from how UI updates are applied, there's flexibility in how Razor Components can be hosted. ASP.NET Core Razor Components in .NET Core 3.0 adds support for hosting Razor Components on the server in an ASP.NET Core app. UI updates are handled over a SignalR connection.
+
+The runtime:
+
+* Handles sending UI events from the browser to the server.
+* Applies UI updates sent by the server back to the browser after running the components.
+
+The connection used by Razor Components to communicate with the browser is also used to handle JavaScript interop calls.
+
+![Razor Components runs .NET code on the server and interacts with the Document Object Model on the client over a SignalR connection](index/_static/aspnet-core-razor-components.png)
+
+For more information, see <xref:razor-components/hosting-models#server-side-hosting-model>.
+
+### Client-side hosting model
+
+Running .NET code inside web browsers is made possible by a relatively new technology, [WebAssembly](http://webassembly.org) (abbreviated *wasm*). WebAssembly is an open web standard and supported in web browsers without plugins. WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed.
+
+WebAssembly code can access the full functionality of the browser via JavaScript interop. At the same time, WebAssembly code runs in the same trusted sandbox as JavaScript to prevent malicious actions on the client machine.
+
+![Blazor runs .NET code in the browser with WebAssembly.](index/_static/blazor.png)
+
+When a Blazor app is built and run in a browser:
+
+* C# code files and Razor files are compiled into .NET assemblies.
+* The assemblies and the .NET runtime are downloaded to the browser.
+* Blazor uses JavaScript to bootstrap the .NET runtime and configures the runtime to load assemblies. Document Object Model (DOM) manipulation and browser API calls are handled by the Blazor runtime via JavaScript interop.
+
+To support older browsers that don't support WebAssembly, use the [server-side hosting model](#server-side-hosting-model).
+
+For more information, see <xref:razor-components/hosting-models#client-side-hosting-model>.
+
 ## Infrastructure
 
-Razor Components offers the core facilities that most apps require, including:
+Core facilities that most apps require are provided by the framework, including:
 
 * Layouts
 * Routing
 * Dependency injection
 
-All of these features are optional. When one of these features isn't used in an app, the implementation is stripped out of the app when published by the [Intermediate Language (IL) Linker](xref:host-and-deploy/razor-components/configure-linker).
+These scenarios are optional. When one of these scenarios isn't used by a Blazor app, the implementation is stripped out of the app when it's published by the [Intermediate Language (IL) Linker](xref:host-and-deploy/razor-components/configure-linker).
 
 ## Code sharing and .NET Standard
```

Comment:
This section on code sharing and .NET Standard is really specific to Blazor and probably should move to an intro to Blazor topic instead of here.